scripts/changelog-script.sh

```diff
@@ -62,7 +62,7 @@ $RUN_GITHUB_CHANGELOG_GENERATOR \
 --enhancement-labels "kind/feature" \
 --bugs-label "**Bugs:**" \
 --bug-labels "kind/bug" \
---add-sections '{"documentation":{"prefix":"**Documentation:**","labels":["kind/documentation"]}, "tests": {"prefix": "**Testing/CI:**", "labels": ["kind/tests", "kind/failing-test"]}, "cleanup": {"prefix": "**Cleanup/Refactor:**", "labels": ["kind/cleanup"]}}'
+--add-sections '{"documentation":{"prefix":"**Documentation:**","labels":["area/documentation"]}, "tests": {"prefix": "**Testing/CI:**", "labels": ["kind/tests"]}, "cleanup": {"prefix": "**Cleanup/Refactor:**", "labels": ["kind/code-refactoring"]}}'
```

Comment:
Until we delete the `kind/documentation` label (and add the new `area/documentation` one to all issues that had the former label), can we keep both `area/documentation` and `kind/documentation` as a safety measure?

```suggestion
--add-sections '{"documentation":{"prefix":"**Documentation:**","labels":["area/documentation", "kind/documentation"]}, "tests": {"prefix": "**Testing/CI:**", "labels": ["kind/tests"]}, "cleanup": {"prefix": "**Cleanup/Refactor:**", "labels": ["kind/code-refactoring"]}}'
```

Reply:
Do we want to add that label to the former issues? The changelog script only takes into account PRs, no? I see a `--no-issues` flag in the list.

With the new template, all the new PRs will have the required labels, so I don't see why we need both the labels.

Also, I plan on putting this on hold until 3.1.0, so that all the new PRs have the new template, and if not, we can always add them, since they are PRs and much less in number.

Reply:
Whoops - I missed your comment..

> Do we want to add that label to the former issues? The changelog script only takes into account PRs, no? I see a --no-issues flag in the list.
>
You're right in that we are only considering PRs. When I tested manually `github-changelog-generator`, the default behavior was to pick both completed issues and PRs.

> With the new template, all the new PRs will have the required labels, so I don't see why we need both the labels.
> 
> Also, I plan on putting this on hold until 3.1.0, so that all the new PRs have the new template, and if not, we can always add them, since they are PRs and much less in number.
>
In that case, indeed, we don't need both labels. Makes sense.

Reply:
I am confused - what's the purpose of your latest commit [ed935b4](https://github.com/redhat-developer/odo/pull/6219/commits/ed935b4e4eb3c78d09756fb293129a2e5d5f94bb) then?

Reply:
> I am confused - what's the purpose of your latest commit [ed935b4](https://github.com/redhat-developer/odo/pull/6219/commits/ed935b4e4eb3c78d09756fb293129a2e5d5f94bb) then?

Sorry about that, I did not make myself clear.
I was wondering...
With the original change, if we use `/area documentation` in combination with `/kind bug` and `/kind feature`, then the documentation related PRs will show up in the bugs and features list _and_ in the documentation list when you generate the changelog, you would've have had to weed them out from features/bugs list to avoid duplicates, it didn't seem to make sense to me.
So I figured, it'll make much more sense to simply assign all documentation related issues `/kind documentation`.
WDYT? Or was the original change more acceptable?

Reply:
Reverted the changes. @rm3l can you please take a look at it again? I would like to merge it and start using it as soon as possible.

Reply:
Okay, so I just run the changelog script with your changes, and here is an example of the output:

<details>
<summary>./scripts/changelog-script.sh v3.0.0 v3.1.0</summary>

```
# Release of v3.1.0

## [v3.1.0](https://github.com/redhat-developer/odo/tree/v3.1.0) (2022-10-20)

[Full Changelog](https://github.com/redhat-developer/odo/compare/v3.1.0...v3.10)

**Features/Enhancements:**

- Add a link to advanced guide in quickstart guides [\#6228](https://github.com/redhat-developer/odo/pull/6228) ([valaparthvi](https://github.com/valaparthvi))

**Documentation:**

- Update message about v3 and version selector in documentation [\#6238](https://github.com/redhat-developer/odo/pull/6238) ([rm3l](https://github.com/rm3l))
- Update version to 3.0.0 in doc and `build/VERSION` file [\#6220](https://github.com/redhat-developer/odo/pull/6220) ([rm3l](https://github.com/rm3l))
- odo 3.0.0 GA blog [\#6188](https://github.com/redhat-developer/odo/pull/6188) ([valaparthvi](https://github.com/valaparthvi))

**Testing/CI:**

- fix test failure due to devfile-registry  [\#6235](https://github.com/redhat-developer/odo/pull/6235) ([anandrkskd](https://github.com/anandrkskd))
- enable quarkus e2e test [\#6224](https://github.com/redhat-developer/odo/pull/6224) ([anandrkskd](https://github.com/anandrkskd))

**Cleanup/Refactor:**

- Build context as part of GenericRun [\#6202](https://github.com/redhat-developer/odo/pull/6202) ([feloy](https://github.com/feloy))

**Merged pull requests:**

- Minor fixes to documentation [\#6230](https://github.com/redhat-developer/odo/pull/6230) ([valaparthvi](https://github.com/valaparthvi))

## [v3.1.0](https://github.com/redhat-developer/odo/tree/v3.1.0) (2022-10-18)

[Full Changelog](https://github.com/redhat-developer/odo/compare/v3.0.0...v3.1.0)

**Features/Enhancements:**

- Allow passing `caller` to telemetry [\#6223](https://github.com/redhat-developer/odo/pull/6223) ([rm3l](https://github.com/rm3l))
- Move --var-file and --var as generic flags [\#6201](https://github.com/redhat-developer/odo/pull/6201) ([feloy](https://github.com/feloy))
- Support a generic --run-on flag [\#6187](https://github.com/redhat-developer/odo/pull/6187) ([feloy](https://github.com/feloy))

**Bugs:**

- Fix broken links in the installation guide [\#6214](https://github.com/redhat-developer/odo/pull/6214) ([valaparthvi](https://github.com/valaparthvi))
- Fix help and doc for `odo completion zsh` [\#6167](https://github.com/redhat-developer/odo/pull/6167) ([rm3l](https://github.com/rm3l))
- Fix message for interactive odo delete project [\#6154](https://github.com/redhat-developer/odo/pull/6154) ([valaparthvi](https://github.com/valaparthvi))
- Fix panic when component name is long but not app [\#6136](https://github.com/redhat-developer/odo/pull/6136) ([feloy](https://github.com/feloy))

**Documentation:**

- Update USAGE\_DATA.md [\#6182](https://github.com/redhat-developer/odo/pull/6182) ([valaparthvi](https://github.com/valaparthvi))
- Adjust sidebars on odo.dev website [\#6156](https://github.com/redhat-developer/odo/pull/6156) ([valaparthvi](https://github.com/valaparthvi))
- Make command reference docs consistent \(2\) [\#6153](https://github.com/redhat-developer/odo/pull/6153) ([valaparthvi](https://github.com/valaparthvi))
- Document how `odo` translates `container` component into Kubernetes resources [\#6145](https://github.com/redhat-developer/odo/pull/6145) ([rm3l](https://github.com/rm3l))
- Make command reference docs consistent [\#6126](https://github.com/redhat-developer/odo/pull/6126) ([valaparthvi](https://github.com/valaparthvi))
- Add documentation on connecting to a service [\#6114](https://github.com/redhat-developer/odo/pull/6114) ([valaparthvi](https://github.com/valaparthvi))

**Testing/CI:**

- Fix postgreSQL containers failing due to file permissions errors [\#6190](https://github.com/redhat-developer/odo/pull/6190) ([feloy](https://github.com/feloy))
- suspend quarkus test [\#6184](https://github.com/redhat-developer/odo/pull/6184) ([anandrkskd](https://github.com/anandrkskd))
- Revert devfile python ID [\#6164](https://github.com/redhat-developer/odo/pull/6164) ([valaparthvi](https://github.com/valaparthvi))

**Merged pull requests:**

- Bump odo to v3.1.0 [\#6231](https://github.com/redhat-developer/odo/pull/6231) ([dharmit](https://github.com/dharmit))
- Removal from OWNERS [\#6218](https://github.com/redhat-developer/odo/pull/6218) ([cdrage](https://github.com/cdrage))
- Merge `v3.0.0` tag into `main` [\#6216](https://github.com/redhat-developer/odo/pull/6216) ([rm3l](https://github.com/rm3l))
- Add GitHub Workflow for issue management [\#6209](https://github.com/redhat-developer/odo/pull/6209) ([rm3l](https://github.com/rm3l))
- Update Algolia settings [\#6203](https://github.com/redhat-developer/odo/pull/6203) ([dharmit](https://github.com/dharmit))
- Note regarding image visibility for odo deploy [\#6191](https://github.com/redhat-developer/odo/pull/6191) ([dharmit](https://github.com/dharmit))
- Refactor remotecmd [\#6169](https://github.com/redhat-developer/odo/pull/6169) ([feloy](https://github.com/feloy))
- Refacto sync pkg [\#6163](https://github.com/redhat-developer/odo/pull/6163) ([feloy](https://github.com/feloy))
- Blog for v3 rc2 release [\#6161](https://github.com/redhat-developer/odo/pull/6161) ([dharmit](https://github.com/dharmit))
- Add CLI images / information on index page of odo.dev [\#6148](https://github.com/redhat-developer/odo/pull/6148) ([cdrage](https://github.com/cdrage))
- Use Deploying an Application not Deploying application [\#6144](https://github.com/redhat-developer/odo/pull/6144) ([cdrage](https://github.com/cdrage))
- Refacto odo dev [\#6141](https://github.com/redhat-developer/odo/pull/6141) ([feloy](https://github.com/feloy))
-  Add extra output when init project using odo dev or deploy [\#6127](https://github.com/redhat-developer/odo/pull/6127) ([cdrage](https://github.com/cdrage))
- Create document highlighting dev and deploy [\#6120](https://github.com/redhat-developer/odo/pull/6120) ([cdrage](https://github.com/cdrage))

\* *This Changelog was automatically generated by [github_changelog_generator](https://github.com/github-changelog-generator/github-changelog-generator)*

```

</details>

There were concerns about possible duplicate items in multiple sections, but this does not seem to happen (there is an open feature request for that in https://github.com/github-changelog-generator/github-changelog-generator/issues/978).

That said, I am not sure how the changelog generator decides which section an item should be put in if there are multiple possibilities.

Also, the behavior is essentially similar to what we already have.
I checked the latest release notes, and PRs with multiple labels (such as both `kind/bug` and `kind/documentation`) show up in a single section, not necessarily Documentation.

So to me, this is fine. We just need to make sure that Documentation PRs only have `area/documentation` label and no `kind/*` labels, maybe by mentioning this in `.github/PULL_REQUEST_TEMPLATE.md`.